tests/integrations/android_sdk/test_verify_android_sdk.py

```diff
@@ -0,0 +1,281 @@
+        assert os.access(str(example_tool), os.X_OK)
+
+    # The license has been accepted
+    assert (root_path / "licenses" / "android-sdk-license").exists()
+
+    # The returned SDK has the expected root path.
+    assert sdk.root_path == root_path
+
+
+@pytest.mark.skipif(
+    sys.platform == "win32", reason="executable permission doesn't make sense on Windows"
+)
+def test_download_sdk_if_sdkmanager_not_executable(mock_command, tmp_path):
+    """An SDK will be downloaded and unpackged if `tools/bin/sdkmanager` exists
+    but does not have its permissions set properly."""
+    root_path = tmp_path / ".briefcase" / "tools" / "android_sdk"
+
+    # Create non-executable `sdkmanager`.
+    root_path = tmp_path / ".briefcase" / "tools" / "android_sdk"
+    (root_path / "tools" / "bin").mkdir(parents=True)
+    (root_path / "tools" / "bin" / "sdkmanager").touch(mode=0o644)
+
+    # The download will produce a cached file
+    cache_file = MagicMock()
+    cache_file.__str__.return_value = "/path/to/download.zip"
+    mock_command.download_url.return_value = cache_file
+
+    # Set up a side effect for accepting the license
+    mock_command.subprocess.run.side_effect = accept_license(root_path)
+
+    # Call `verify_android_sdk()`
+    sdk = verify_android_sdk(mock_command)
+
+    # Validate that the SDK was downloaded and unpacked
+    mock_command.download_url.assert_called_once_with(
+        url="https://dl.google.com/android/repository/sdk-tools-unknown-4333796.zip",
+        download_path=mock_command.dot_briefcase_path / "tools",
+    )
+    mock_command.shutil.unpack_archive.assert_called_once_with(
+        "/path/to/download.zip",
+        extract_dir=str(root_path)
+    )
+
+    # The cached file will be deleeted
+    cache_file.unlink.assert_called_once_with()
+
+    # The license has been accepted
+    assert (root_path / "licenses" / "android-sdk-license").exists()
+
+    # The returned SDK has the expected root path.
+    assert sdk.root_path == root_path
+
+
+def test_raises_networkfailure_on_connectionerror(mock_command):
+    "If an error occurs downloading the ZIP file, and error is raised."
+    mock_command.download_url.side_effect = requests_exceptions.ConnectionError()
+
+    with pytest.raises(NetworkFailure):
+        verify_android_sdk(mock_command)
+
+    # The download was attempted
+    mock_command.download_url.assert_called_once_with(
+        url="https://dl.google.com/android/repository/sdk-tools-unknown-4333796.zip",
+        download_path=mock_command.dot_briefcase_path / "tools",
+    )
+    # But no unpack occurred
+    assert mock_command.shutil.unpack_archive.call_count == 0
+
+
+def test_detects_bad_zipfile(mock_command, tmp_path):
+    "If the ZIP file is corrupted, an error is raised."
+    root_path = tmp_path / ".briefcase" / "tools" / "android_sdk"
+
+    # The download will produce a cached file
+    cache_file = MagicMock()
+    cache_file.__str__.return_value = "/path/to/download.zip"
```

Comment:
Same remark as elsewhere about weird mocking-out `__str__`.

Might make sense to move this to a pytest fixture, if we you have to keep doing this weird thing.

Reply:
I'll keep that in mind.